src/components/NewGalaxyModal.js

```diff
@@ -61,17 +79,38 @@ export const NewGalaxyModal = _.flow(
       ['launchWarn', () => {
         return h(GalaxyLaunchButton, { app, onClick: onDismiss })
       }],
+      ['paused', () => {
+        return h(Fragment, [
+          h(ButtonPrimary, { style: { marginRight: 'auto' }, onClick: () => setViewMode('deleteWarn') }, ['Delete']),
+          h(ButtonSecondary, { style: { marginRight: '1rem' }, onClick: () => { resumeGalaxy() } }, ['Resume'])
+        ])
+      }],
       [Utils.DEFAULT, () => {
         return !!app ?
           h(Fragment, [
             h(ButtonSecondary, { style: { marginRight: 'auto' }, onClick: () => setViewMode('deleteWarn') }, ['Delete']),
-            h(ButtonPrimary, { onClick: () => setViewMode('launchWarn') }, ['Launch Galaxy'])
+            (app.status === 'RUNNING') && h(ButtonSecondary, { style: { marginRight: '1rem' }, onClick: () => { pauseGalaxy() } }, ['Pause']),
+            (app.status === 'STOPPED') && h(ButtonPrimary, { style: { marginRight: '1rem' }, onClick: () => { resumeGalaxy() } }, ['Resume']),
+            (app.status === 'RUNNING') && h(ButtonPrimary, { onClick: () => setViewMode('launchWarn') }, ['Launch Galaxy'])
           ]) :
           h(ButtonPrimary, { onClick: () => setViewMode('createWarn') }, ['Next'])
       }]
     )
   }
 
+  const renderMessaging = () => {
+    return Utils.cond(
+      [viewMode === 'createWarn', renderCreateWarning],
+      [viewMode === 'deleteWarn', renderDeleteWarning],
+      [viewMode === 'launchWarn', renderLaunchWarning],
+      [!!app && (app.status === 'STOPPED'), renderPaused],
+      [!!app && (app.status === 'STOPPING'), renderStoppingWarning],
+      [!!app && (app.status === 'STARTING'), renderStartingWarning],
+      [Utils.DEFAULT, renderDefaultCase]
```

Comment:
TOL: There isn't anything inherently wrong with this code, but a couple thoughts on the structure:

Currently we're using `viewMode` as a nice top-level organizing principle to represent the multiple distinct 'modes' we flip between based on user interaction. Rather than mixing in the status-based logic here, we could move it into the 'default' mode, creating a two-level logical hierarchy. Conceptually, that would say "This dialog has multiple 'modes' that we switch between. Additionally, the default mode (that we see when it first opens) might vary based on the app's current status."

What I like about that framing is that it paves the way for some UX questions. For example, what's the reason for hiding the machine details when starting/stopping? We could easily continue to provide that info, and also show the current status.

Reply:
This sounds good. So just to make sure I understand, would this be the same as the renderPaused comment you mentioned above? Where we would move the app.status logic into the renderDefaultCase block?

Reply:
Yep, they're overlapping suggestions. If we go with the re-organization discussed here, then the similar code can be naturally combined as part of that.